File: 백준/Silver/1735. 분수 합/분수 합.py
def GCD(a, b):
    while b != 0:
        a, b = b, a%b
    return a
x=list(map(int,input().split()))
y=list(map(int,input().split()))
a=y[0]*x[1]+x[0]*y[1]
b=x[1]*y[1]
c=GCD(a,b)
print(a//c,b//c)

File: 백준/Silver/1735. 분수 합/분수 합.java
import java.util.*;
import java.io.*;
public class Main {
    public static int gcd(int a, int b){
        if(a== 0) return b;
        else return gcd(b%a, a);
    }
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int[] a = new int[2];
        for(int i = 0; i < 2; i++){
            a[i]= Integer.parseInt(st.nextToken());
        }
        st = new StringTokenizer(br.readLine());
        int[] b = new int[2];
        for(int i = 0; i < 2; i++){
            b[i]= Integer.parseInt(st.nextToken());
        }
        int c= a[0]*b[1]+ b[0]* a[1];
        int d= a[1]*b[1];
        int mul= gcd(Math.min(c,d), Math.max(c,d));
        System.out.printf("%d %d", c/mul, d/mul);
    }
}